app/api/server/lib/email-channel.js

```diff
@@ -0,0 +1,63 @@
+import { EmailChannel as EmailChannelRaw } from '../../../models/server/raw';
+import { EmailChannel, Users } from '../../../models';
+import { hasPermissionAsync } from '../../../authorization/server/functions/hasPermission';
+
+export async function find({ userId, query = {}, pagination: { offset, count, sort } }) {
+	if (!await hasPermissionAsync(userId, 'manage-email-channels')) {
+		throw new Error('error-not-allowed');
+	}
+	const cursor = EmailChannelRaw.find(query, {
+		sort: sort || { name: 1 },
+		skip: offset,
+		limit: count,
+	});
+
+	const total = await cursor.count();
+
+	const emailChannels = await cursor.toArray();
+
+	return {
+		emailChannels,
+		count: emailChannels.length,
+		offset,
+		total,
+	};
+}
+
+export async function findOne({ userId, id }) {
+	if (!hasPermissionAsync(userId, 'manage-email-channels')) {
+		throw new Error('error-not-allowed');
+	}
+	return EmailChannel.findOneById(id);
+}
+
+export async function createEdit(emailChannelParams, userId) {
+	if (!hasPermissionAsync(userId, 'manage-email-channels')) {
+		throw new Error('error-not-allowed');
+	}
+
+	const updateEmailChannel = {
+		$set: { },
+	};
+
+	const { _id, active, name, email, description, senderInfo, department, smtp, imap } = emailChannelParams;
+
+	if (!_id) {
+		emailChannelParams.ts = new Date();
+		emailChannelParams._createdBy = Users.findOne(this.userId, { fields: { username: 1 } });
```

Comment:
This implementation is not working propertly.
I created a new email channel logged with my user and `rocket.chat` user has been stored as `_createdBy`.

Reply:
I'm trying to reproduce but its going well here.